lib/Sema/TypeCheckConstraints.cpp

```diff
@@ -451,6 +451,37 @@ resolveDeclRefExpr(UnresolvedDeclRefExpr *UDRE, DeclContext *DC) {
       diagnose(Loc, diag::use_unresolved_identifier, Name,
                UDRE->getName().isOperator())
         .highlight(Loc);
+      const char *buffer = Name.get();
```

Comment:
I think we do want a similar error here, but
(a) the code duplication makes me unhappy, and
(b) if someone actually does define an unusual operator with multiple characters in it, being warned about each of them individually seems suboptimal.

Maybe a better answer would be to "deconfuse" the entire operator identifier and tell the user to try that, with special-case wording for single-character operators. What do you think?